sdk/openai/azure-ai-openai/src/test/java/com/azure/ai/openai/OpenAIClientBuilderTest.java

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.*;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Configuration;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Mono;
+import java.lang.reflect.Field;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class OpenAIClientBuilderTest {
+
+    @Test
+    void testBuildClientWithKeyCredential() {
+        OpenAIClientBuilder builder = new OpenAIClientBuilder().credential(new AzureKeyCredential("fake-key"))
+            .endpoint("https://api.openai.com/");
+
+        assertNotNull(builder.buildClient());
+    }
+
+    @Test
+    void testBuildClientWithTokenCredential() throws NoSuchFieldException, IllegalAccessException {
+        TokenCredential tokenCredential = (request) -> {
+            assertNotNull(request, "The request parameter should not be null");
+            assertTrue(request.getScopes().isEmpty(), "The request scopes should be empty");
+            return Mono.just(new AccessToken("fake-token", OffsetDateTime.now().plusHours(2)));
+        };
+
+        OpenAIClientBuilder builder
+            = new OpenAIClientBuilder().credential(tokenCredential).endpoint("https://api.openai.com/");
+
+        OpenAIClient client = builder.buildClient();
+        assertNotNull(client, "The client should not be null");
+
+        // Access private fields using reflection
+        Field endpointField = OpenAIClientBuilder.class.getDeclaredField("endpoint");
```

Comment:
We may want to be a bit more careful here, most of the methods and member variables in `OpenAIClientBuilder` are generated so they are not succeptible to change much, I think. But in the given case that `endpoint` no longer exists as such, what would be the behaviour of this test? Is this when `NoSuchFieldException` is thrown?